doc/specs/stdlib_linalg.md

```diff
@@ -239,33 +239,34 @@ Pure function.
 
 ### Description
 
-Construct the identity matrix.
+Constructs the identity matrix.
 
 ### Syntax
 
-`I = ` [[stdlib_linalg(module):eye(function)]] `(dim1 [, dim2])`
+`I = ` [[stdlib_linalg(module):eye(function)]] `(dim1 [, dim2] [, mold])`
 
 ### Arguments
 
-`dim1`: Shall be a scalar of default type `integer`.
-This is an `intent(in)` argument. 
-
-`dim2`: Shall be a scalar of default type `integer`.
-This is an `intent(in)` and `optional` argument. 
+- `dim1`: A scalar of type `integer`. This is an `intent(in)` argument and specifies the number of rows.
+- `dim2`: A scalar of type `integer`. This is an optional `intent(in)` argument specifying the number of columns. If not provided, the matrix is square (`dim1 = dim2`).
+- `mold`: A scalar of any supported `integer`, `real`, or `complex` type. This is an optional `intent(in)` argument. If provided, the returned identity matrix will have the same type and kind as `mold`. If not provided, the matrix will be of type `integer(int8)` by default.
```

Comment:
Based on the Fortran Discourse thread, is it still wise to return `integer(int8)` by default (instead of `integer`)? (the API is still experimental and so could still be changed).

Reply:
Thanks a lot @jvdp1 for the review. I left default behavior unchanged to not introduce API changes. However, my personal preference would be that the default return type should be `real(real32)` or `real(real64)`. The reason is that we've all made this Fortran mistake of integer division far too many times :) So I think that in the interest of safety for the least experienced Fortran programmers, it would be safe(r) to use a `real` return value.

Reply:
I agree with you: I also think it would be better to return a `real` identity matrix. The `mold` argument will let the user to return another type and kind if desired.